handler/OpenIDLoginHandler.php

```diff
@@ -0,0 +1,281 @@
+            }
+            
+            $context = Application::get()->getRequest()->getContext();
+            $contextId = ($context == null) ? PKPApplication::CONTEXT_SITE : $context->getId();
+            $settingsJson = $this->plugin->getSetting($contextId, 'openIDSettings');
+
+            if (isset($user)) {
+                $user->setData('openid::lastProvider', null);
+                Repo::user()->edit($user);
+            }
+
+            $sessionManager = SessionManager::getManager();
+            $session = $sessionManager->getUserSession();
+            $token = $session->getSessionVar('token');
+
+            Validation::logout();
+            if (isset($settingsJson, $lastProvider)) {
+                $providerList = json_decode($settingsJson, true)['provider'];
+                $settings = $providerList[$lastProvider];
+
+                if (!empty($settings['logoutUrl']) && isset($settings['clientId'])) {
+                    $request->redirectUrl(
+                        $settings['logoutUrl'] .
+                        '?client_id=' . urlencode($settings['clientId']) .
+                        '&post_redirect_uri=' . urlencode($router->url($request, $context->getPath(), "index")) .
+                        '&id_token_hint=' . urlencode($token)
+                    );
+                }
+            }
+        }
+        $request->redirect(Application::get()->getRequest()->getContext()->getPath(), 'index');
+    }
+
+    /**
+     * Sets user friendly error messages, which are thrown during the OpenID auth process.
+     */
+    private function _setSSOErrorMessages(string $ssoError, TemplateManager $templateMgr, Request $request)
+    {
+        $templateMgr->assign('openidError', true);
+        switch ($ssoError) {
+            case 'connect_data':
+                $templateMgr->assign('errorMsg', 'plugins.generic.openid.error.openid.connect.desc.data');
+                break;
+            case 'connect_key':
+                $templateMgr->assign('errorMsg', 'plugins.generic.openid.error.openid.connect.desc.key');
+                break;
+            case 'cert':
+                $templateMgr->assign('errorMsg', 'plugins.generic.openid.error.openid.cert.desc');
+                break;
+            case 'disabled':
+                $reason = $request->getUserVar('sso_error_msg');
+                $templateMgr->assign('accountDisabled', true);
+                if (isset($reason) && !empty($reason)) {
+                    $templateMgr->assign('errorMsg', 'plugins.generic.openid.error.openid.disabled.with');
+                    $templateMgr->assign('reason', $reason);
+                } else {
+                    $templateMgr->assign('errorMsg', 'plugins.generic.openid.error.openid.disabled.without');
+                }
+                break;
+        }
+
+        $context = $request->getContext();
+        $supportEmail = $context != null ? $context->getSetting('supportEmail') : null;
```

Comment:
This will leave the support email blank when the context isn't set; I think we could use the site-wide contact address in that case.

Reply:
I changed that to `$supportEmail = $context != null ? $context->getSetting('supportEmail') : $request->getSite()->getLocalizedContactEmail();`

Is this what you are referring to?